src/OpenTelemetry.Exporter.ZPages/Implementation/ZPagesSpans.cs

```diff
@@ -0,0 +1,122 @@
+            ProcessingSpanList = new Dictionary<string, long>();
+            CurrentMinuteSpanList = new Dictionary<string, ZPagesSpanInformation>();
+            CurrentHourSpanList = new Dictionary<string, ZPagesSpanInformation>();
+            TotalSpanCount = new Dictionary<string, long>();
+            TotalEndedSpanCount = new Dictionary<string, long>();
+            TotalSpanErrorCount = new Dictionary<string, long>();
+            TotalSpanLatency = new Dictionary<string, long>();
+            startTime = DateTimeOffset.Now.ToUnixTimeMilliseconds();
+        }
+
+        /// <summary>
+        /// Gets or sets ZQueue which stores the minute-wise span information.
+        /// </summary>
+        public static LinkedList<Dictionary<string, ZPagesSpanInformation>> ZQueue { get; set; }
+
+        /// <summary>
+        /// Gets or sets the current minute span information list.
+        /// </summary>
+        public static Dictionary<string, ZPagesSpanInformation> CurrentMinuteSpanList { get; set; }
+
+        /// <summary>
+        /// Gets or sets the current hour span information list.
+        /// </summary>
+        public static Dictionary<string, ZPagesSpanInformation> CurrentHourSpanList { get; set; }
+
+        /// <summary>
+        /// Gets or sets the processing span information list. This holds the names of the spans which have not ended yet, along with the active count.
+        /// </summary>
+        public static Dictionary<string, long> ProcessingSpanList { get; set; }
+
+        /// <summary>
+        /// Gets or sets the count of spans name-wise.
+        /// </summary>
+        public static Dictionary<string, long> TotalSpanCount { get; set; }
+
+        /// <summary>
+        /// Gets or sets the count of ended spans name-wise.
+        /// </summary>
+        public static Dictionary<string, long> TotalEndedSpanCount { get; set; }
+
+        /// <summary>
+        /// Gets or sets the count of span errors according to span name.
+        /// </summary>
+        public static Dictionary<string, long> TotalSpanErrorCount { get; set; }
+
+        /// <summary>
+        /// Gets or sets the total latency of spans name-wise.
+        /// </summary>
+        public static Dictionary<string, long> TotalSpanLatency { get; set; }
+
+        /// <summary>
+        /// Gets or sets the retention time (in milliseconds) for the metrics.
+        /// </summary>
+        public static long RetentionTime { get; set; }
+
+        /// <summary>
+        /// Triggers Calculations every minute.
+        /// </summary>
+        /// <param name="source">Source.</param>
+        /// <param name="e">Event Arguments.</param>
+        public static void TriggerMinuteCalculations(object source, ElapsedEventArgs e)
+        {
+            // Enqueue the last minute's span information list
+            ZQueue.AddFirst(CurrentMinuteSpanList);
+            CurrentMinuteSpanList.Clear();
```

Comment:
I don't understand. You are cleaning the dictionary that have just been queued for aggregation... So you wouldn't be aggregating anything at all

Reply:
When a new minute starts, the dictionary gets cleared. So it will start storing information only about new spans. If the queued information needs to be accessed, it is at the front of the queue.